File: 프로그래머스/2/135807. 숫자 카드 나누기/숫자 카드 나누기.py
def solution(arrayA, arrayB):
    answer = 0

    def gcd(a, b):
        while b > 0:
            a, b = b, a % b
        return a
    
    A = arrayA[0]
    for i in arrayA[1:]:
        A = gcd(A,i)
        
    B = arrayB[0]
    for i in arrayB[1:]:
        B = gcd(B,i)
    
    for i in arrayB:
        if i%A==0:
            A=0
            break
    for i in arrayA:
        if i%B==0:
            B=0
            break
            
    answer = max(A,B)
    return answer

File: 프로그래머스/2/135807. 숫자 카드 나누기/숫자 카드 나누기.js
function solution(arrayA, arrayB) {
    let a = getDvsr(arrayA, arrayB);
    let b = getDvsr(arrayB, arrayA);
    return Math.max(a,b);
}
function getDvsr(arrayA, arrayB){
    let answer = 0;
    const gcd =  (a, b) => a % b === 0 ? b : gcd(b, a % b);
    arrayA.forEach(val => answer = gcd(answer, val));
    if(arrayB.some(val => val%answer === 0)) return 0;
    
    return answer;
}